### PR TITLE
feat(OnyxSelect): Implement OnyxSelect open direction and alignment

### PR DESCRIPTION
**Pull request by MajaZarkova**
Relates to #1198 

Added new feature which is calculating the space above and below the component and based on that the flyout will open to the top or bottom. I also added new 'alignment' property which will align the flyout to the right, left or will set the width to match the width of the input.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] If a new component is added, at least one [Playwright screenshot test](https://github.com/SchwarzIT/onyx/actions/workflows/playwright-screenshots.yml) is added
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: .changeset/ten-scissors-boil.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": minor
+---
+
+Added new feature which is calculating the space above and below the component and based on that the flyout will open to the top or bottom. I also added new 'alignment' property which will align the flyout to the right, left or will set the width to match the width of the input.
```
Comment (larsrickert) [inline .changeset/ten-scissors-boil.md:5]: feat(OnyxSelect): add property `alignment` and auto detection for opening to top/bottom depending on the available space

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/src/styles/mixins/list.scss [hunk 2]
```diff
@@ -6,9 +6,9 @@
   padding: $wrapper-padding 0;
   box-shadow: var(--onyx-shadow-medium-bottom);
   box-sizing: border-box;
-  width: max-content;
+  // width: max-content;
   min-width: var(--onyx-spacing-4xl);
-  max-width: 20rem;
+  // max-width: 20rem;
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/src/styles/mixins/list.scss]: I set the max-width when left/right is used and I will clarify with Jannick about full width

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: I checked Storybook and it's hovered there as well 

image

Did we change something in the button?

[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: There is --onyx-color-base-background-tinted applied to the button

[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: https://github.com/SchwarzIT/onyx/pull/1609/files

[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: Here we changed the color of the button

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue [hunk 3]
```diff
@@ -83,6 +84,8 @@ const { state: open } = useManagedState(
 );
 
 const selectRef = ref<HTMLElement>();
+const flyoutRef = ref<HTMLElement>();
+const openDirection = ref<"top" | "bottom">();
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue]: This is handled by the new composable now

File: packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue [hunk 4]
```diff
@@ -178,6 +181,44 @@ const onToggle = async (preventFocus?: boolean) => {
       miniSearch.value?.focus();
     }
   }
+
+  detectOpenDirection();
```
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/src/components/OnyxSelect/OnyxSelect.vue]: This is handled by the composable

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png
[the remark this answers is not in the document]
Comment (larsrickert) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: Ups, thats not intentional then and a bug...

File: packages/sit-onyx/src/styles/mixins/list.scss
Comment (larsrickert) [inline packages/sit-onyx/src/styles/mixins/list.scss:1]: Removing the max width here would also remove it for all other flyouts like nav item, user menu etc...I think its better if we leave it here and override/unset it only in the select if full width is used

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png
[the remark this answers is not in the document]
Comment (MajaZarkova) [reply packages/sit-onyx/playwright/snapshots/components/OnyxSelect/Select-loading--chromium-linux.png:1]: Follow up ticket -> #1764

**Review by larsrickert — APPROVED** [review]

**Pull request merged**